src/HazelcastClient.ts

```diff
@@ -44,8 +44,8 @@ class HazelcastClient {
             return this.partitionService.initialize();
         }).then(() => {
             deferred.resolve(this);
-        }).catch(() => {
-            deferred.reject('Client failed to start');
+        }).catch((e) => {
+            deferred.reject(e);
```

Comment:
Propagating the error to upper level is already default behaviour. This `.catch` is unnecessary. Correct me if I am wrong.